---
title: "$F_{ST}$ variation across human traits"
author: "Ian Brettell"
date: '`r format(Sys.Date())`'
output:
  html_document:
    toc: true
    toc_float: true
    dev: 'svg'
    number_sections: true
    keep_md: false
    pandoc_args: --lua-filter=color-text.lua
    highlight: pygments
#output: html_notebook
#editor_options: 
#  chunk_output_type: inline
---

NOTE: add to R container:

* plotly
* DT: `remotes::install_github('rstudio/DT')`


* [Working directory]{color="#4f0943"} on EBI Codon HPC: `/hps/nobackup/birney/users/ian/hmn_fst`
* [GitHub repository]{color="#4f0943"}: <https://github.com/brettellebi/human_traits_fst>


# Source libraries, functions and plotting parameters

```{r, warning = F, message = F}
library(here)

source(here::here("code/scripts/20210628_source.R"))
```

# Get full list of traits, associations and studies from the GWAS Catalog

## Traits

```{r, eval = F}
date_of_collection = "20210628"
long_term_dir = "/nfs/research/birney/users/ian/hmn_fst/gwasrapidd"
out_file = here::here("code/snakemake/20210625/config",
                      paste(date_of_collection, "_all_traits.tsv", sep = ""))

# Get list of all traits in database
all_traits = gwasrapidd::get_traits()

# Filter out IDs that cause an error
all_traits_tbl = all_traits@traits %>% 
  dplyr::filter(efo_id != "CHEBI:7916") %>%
  # Save to file
  readr::write_tsv(out_file)
```

```{r, include = F}
date_of_collection = "20210628"
out_file = here::here("code/snakemake/20210625/config",
                      paste(date_of_collection, "_all_traits.tsv", sep = ""))

all_traits_tbl = readr::read_tsv(out_file)
```

## Associations

**NOTE**: This stage is included in the Snakemake process linked below.

```{r, eval = F}
#Need to do them sequentially because the GWAS Catalog can't handle dozens (let alone thousands) of simultaneous queries.
date_of_collection = "20210628"
long_term_dir = "/nfs/research/birney/users/ian/hmn_fst/gwasrapidd"


counter = 0
lapply(all_traits@traits$efo_id, function(EFO_ID) {
  counter <<- counter + 1
  # set output file name
  out_path = file.path(long_term_dir,
                       date_of_collection,
                       "associations_raw",
                       paste(EFO_ID, ".rds", sep = ""))
  # if output file doesn't already exist, get associations and save
  if (!file.exists(out_path)){
    out = gwasrapidd::get_associations(efo_id = EFO_ID)
    saveRDS(out, out_path)
  } else {
    print(paste(counter, ". ", EFO_ID, ": File already exists.", sep = ""))
  }
  # save
})
```

## Read in associations and filter for those with more than 50 unique SNPs

```{r, eval = F}
date_of_collection = "20210628"
long_term_dir = "/nfs/research/birney/users/ian/hmn_fst/gwasrapidd"
out_file = here::here("code/snakemake/20210625/config",
                      paste(date_of_collection, "_filtered_traits.tsv", sep = ""))

all_assocs = lapply(all_traits_tbl$efo_id, function(EFO_ID){
  in_path = file.path(long_term_dir,
                      date_of_collection,
                      "associations_raw",
                      paste(EFO_ID, ".rds", sep = ""))
  readRDS(in_path)
})
names(all_assocs) = all_traits_tbl$efo_id

# filter for those with more than 50 unique SNPs
filt_assocs = all_assocs %>% 
  purrr::keep(function(EFO_ID){
    length(unique(EFO_ID@risk_alleles$variant_id)) >= 50
  })

# Filter `all_traits_tbl` for those traits and save to file
filt_traits_tbl = all_traits_tbl %>% 
  dplyr::filter(efo_id %in% names(filt_assocs)) %>% 
  readr::write_tsv(out_file)
```

```{r, include = F}
date_of_collection = "20210628"
out_file = here::here("code/snakemake/20210625/config",
                      paste(date_of_collection, "_filtered_traits.tsv", sep = ""))

all_traits_tbl = readr::read_tsv(out_file)
```

# Run Snakemake pipeline to extract data

Full Snakemake pipeline here: <https://github.com/brettellebi/human_traits_fst/tree/master/code/snakemake/20210625>

# Read in processed data

```{r}
target_dir = "/nfs/research/birney/users/ian/hmn_fst/gwasrapidd/20210628"

# Fst
fst_all_path = file.path(target_dir, "pegas/fst/consol/all.rds")
fst_all = readRDS(fst_all_path) %>%
  lapply(., function(TRAIT_ID){
    # convert to DF
    TRAIT_ID %>%
        data.frame(.) %>% 
        tibble::rownames_to_column(var = "SNP") %>% 
        dplyr::select(SNP, FST_PEGAS = Fst)
  })

# Bind into df
fst_all_df = fst_all %>% 
  dplyr::bind_rows(.id = "EFO_ID") %>% 
  dplyr::left_join(all_traits_tbl %>% 
                     dplyr::select(EFO_ID = efo_id,
                                   TRAIT = trait),
                   by = "EFO_ID")

# Clumped SNPs
## Note that for some traits, no SNPs met the p-value threshold, so Plink1.9 did not produce a `*.clumped` file
## There are therefore fewer files than there are traits
clumped_files = list.files(file.path(target_dir, "plink/clumped"), pattern = "*.clumped", full.names = T)
clumped = lapply(clumped_files, function(FILE){
  readr::read_delim(FILE,
                    delim = " ",
                    col_types = "i-c---------",
                    trim_ws = T)
})
names(clumped) = stringr::str_remove(basename(clumped_files), ".clumped")


# Filter `fst_all` for clumped SNPs
fst_clumped = lapply(names(fst_all), function(TRAIT_ID){
    fst_all[[TRAIT_ID]] %>% 
      dplyr::filter(SNP %in% clumped[[TRAIT_ID]]$SNP)
})
names(fst_clumped) = names(fst_all)
# Bind into DF
fst_clumped_df = fst_clumped %>% 
  dplyr::bind_rows(.id = "EFO_ID") %>% 
  dplyr::left_join(all_traits_tbl %>% 
                     dplyr::select(EFO_ID = efo_id,
                                   TRAIT = trait),
                   by = "EFO_ID")

```

## Add palette based on post-clump SNP count

```{r}
pal_all = turbo(n = length(unique(fst_all_df$EFO_ID)))
names(pal_all) = fst_all_df %>% 
  dplyr::count(TRAIT) %>% 
  dplyr::arrange(desc(n)) %>% 
  dplyr::pull(TRAIT)
```

# SNP counts pre- and post-clumping

```{r, out.width = '150%'}
snp_count_plot = list("PRE-CLUMP" = fst_all_df,
                      "POST-CLUMP" = fst_clumped_df) %>% 
  dplyr::bind_rows(.id = "FILTER") %>% 
  # order variables
  dplyr::mutate(TRAIT = factor(TRAIT, levels = names(pal_all)),
                FILTER = factor(FILTER, levels = c("PRE-CLUMP", "POST-CLUMP"))) %>% 
  ggplot() +
    geom_bar(aes(TRAIT, fill = TRAIT)) +
    scale_fill_manual(values = pal_all) +
    theme_bw() +
    theme(axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          axis.ticks.x = element_blank()) +
    ylab("N LOCI") +
    facet_wrap(~FILTER, nrow = 2) +
    guides(FILL = "none")

plotly::ggplotly(snp_count_plot,
                 tooltip = c("TRAIT", "y"))
```

## Table of post-clump traits

```{r}
fst_clumped_df %>% 
  dplyr::count(EFO_ID, TRAIT) %>% 
  dplyr::rename(N_SNPS = n) %>% 
  dplyr::arrange(desc(N_SNPS)) %>% 
  DT::datatable(., options = list(pageLength = 10))
```


# eCDF for $F_{ST}$

## New palette for clumped traits

```{r}
## Add palette based on post-clump SNP count
pal_clump = turbo(n = length(unique(fst_clumped_df$EFO_ID)))
names(pal_clump) = fst_clumped_df %>% 
  dplyr::count(EFO_ID) %>% 
  dplyr::arrange(desc(n)) %>% 
  dplyr::pull(EFO_ID)

# order `TRAIT` by `EFO_ID` 
od_clumped_traits = fst_clumped_df %>% 
  dplyr::select(EFO_ID, TRAIT) %>% 
  dplyr::distinct() %>% 
  dplyr::left_join(data.frame(EFO_ID = names(pal_clump)),
                   .,
                   by = "EFO_ID") %>% 
  dplyr::pull(TRAIT)

# make names correspond to trait
names(pal_clump) = od_clumped_traits
```

## Plot all

```{r, eval = F}
ecdf_all_faceted = fst_clumped_df %>% 
  dplyr::group_by(EFO_ID) %>%
  # filter for those with >0 SNPs in the clumped dataset
  dplyr::filter(EFO_ID %in% unique(fst_clumped_df$EFO_ID)) %>% 
  dplyr::ungroup() %>% 
  # order
  dplyr::mutate(TRAIT = factor(TRAIT, levels = od_clumped_traits)) %>% 
  ggplot() +
    stat_ecdf(aes(FST_PEGAS, colour = TRAIT)) +
    scale_colour_manual(values = pal_clump) +
    theme_bw() +
    ggtitle("eCDF") +
    facet_wrap(~TRAIT) +
    guides(colour = "none") +
    theme(strip.text.x = element_text(size = 4.5))


ggsave(here::here("plots/20210701_ecdf_all_faceted.png"),
       plot = ecdf_all_faceted, 
       device = "png",
       width = 30,
       height = 22,
       units = "in",
       dpi = 400)

file.copy(here::here("plots/20210701_ecdf_all_faceted.png"), here::here("docs/plots"),
          overwrite = T)
```

```{r, out.width='150%'}
knitr::include_graphics(here::here("docs/plots/20210701_ecdf_all_faceted.png"))
```

## Plotly

### All

```{r, out.width = '175%'}
plotly_ecdf_all = fst_clumped_df %>% 
  # order FST_PEGAS so that it's properly plotted by plotly
  dplyr::arrange(EFO_ID, FST_PEGAS) %>% 
  dplyr::group_by(EFO_ID) %>%
  # filter for those with >0 SNPs in the clumped dataset
  dplyr::filter(EFO_ID %in% unique(fst_clumped_df$EFO_ID)) %>% 
  dplyr::ungroup() %>% 
  # order
  dplyr::mutate(TRAIT = factor(TRAIT, levels = od_clumped_traits)) %>% 
  ggplot() +
    stat_ecdf(aes(FST_PEGAS, colour = TRAIT)) +
    scale_colour_manual(values = pal_clump) +
    theme_bw() +
    ggtitle("eCDF") +
    guides(colour = "none") +
    theme(strip.text.x = element_text(size = 5))

plotly::ggplotly(plotly_ecdf_all,
                 tooltip = c("TRAIT"))
```

### Select

```{r, out.width='150%'}
# get traits to look closer at
target_traits = c("body height",
                  "body mass index",
                  "mathematical ability",
                  "intelligence",
                  "self reported educational attainment",
                  "type ii diabetes mellitus",
                  "asthma",
                  "HIV-1 infection",
                  "viral load", 
                  "heart failure",
                  "diabetes mellitus",
                  "coronary artery disease",
                  "mortality",
                  "longevity",
                  "schziophrenia",
                  "skin pigmentation",
                  "skin pigmentation measurement",
                  "eye colour measurement",
                  "facial morphology measurement",
                  "melanoma",
                  "synophrys measurement",
                  "lip morphology measurement",
                  "squamous cell carcinoma",
                  "tuberculosis",
                  "endometrial carcinoma",
                  "BMI-adjusted hip circumference",
                  "alcohol dependence measurement",
                  "loneliness measurement"
                  )

plotly_ecdf_select = fst_clumped_df %>% 
  # order FST_PEGAS so that it's properly plotted by plotly
  dplyr::arrange(EFO_ID, FST_PEGAS) %>% 
  dplyr::group_by(EFO_ID) %>%
  # filter for those with >0 SNPs in the clumped dataset
  dplyr::filter(EFO_ID %in% unique(fst_clumped_df$EFO_ID)) %>% 
  # filter for `target_traits`
  dplyr::filter(TRAIT %in% target_traits) %>% 
  dplyr::ungroup() %>% 
  # order
  dplyr::mutate(TRAIT = factor(TRAIT, levels = od_clumped_traits)) %>% 
  ggplot() +
    stat_ecdf(aes(FST_PEGAS, colour = TRAIT)) +
    scale_colour_manual(values = pal_clump) +
    theme_bw() +
    ggtitle("eCDF") +
    guides(colour = "none") +
    theme(strip.text.x = element_text(size = 5))

plotly::ggplotly(plotly_ecdf_select,
                 tooltip = c("TRAIT"))
```



